tests/gpflow/test_functions.py

```diff
@@ -371,3 +372,20 @@ def test_models_with_mean_functions_changes(model_class: Type[Any]) -> None:
     assert np.all(var_zero.numpy() == var_non_zero.numpy())
     # predictive mean changes after modifying mean function
     assert not np.all(mu_zero.numpy() == mu_non_zero.numpy())
+
+
+class test_bug_2091_ensureParameterTypeForLinearMeanFunction(unittest.TestCase):
+    """
+    See github issue #2091. This are tests to ensure that the prior is keeped if A is already given as Parameter.
+    """
+
+    # test_parameter_with_correct_shape
+    def test_parameter_with_correct_shape(self) -> None:
+        A = gpflow.Parameter(np.ones((1, 1)), dtype=np.float64)
```

Comment:
You forgot to set a prior on A!

```suggestion
        A = gpflow.Parameter(
            np.ones((1, 1)), dtype=np.float64, prior=tfp.distributions.Normal(0.0, 1.0)
        )
```
 
Also It's useful to include a docstring for each test too, something like
```python
        "Check that Linear copies the prior from a correctly shaped A Parameter"
```